### PR TITLE
Features: asynchronous and 2 options

### PR DESCRIPTION
**Pull request by lukaszmn**
Hi, I added three things to the package:

* ability to work asynchronously
* option to filter messages
* option to hide messages if there are no errors

I also fixed a type `log.success` that displayed `undefined 0 Success` in case of success.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -18,36 +18,51 @@ let message = require('./lib/msg')
...
+  options.filter = options.filter || [];
 
-  return function postHTMLValidate (tree) {
-    w3c.validate({
-      input: render(tree),
-      output: 'json',
-      callback: function (err, res) {
-        res.messages.shift()
-        res.messages.shift()
+  return async function postHTMLValidate (tree) {
```
Comment (Scrum) [inline index.js]: No need to use `async`

**Review by Scrum — CHANGES_REQUESTED** [review]
@lukaszmn Hi, thanks for your input - this is very cool. Could you add a test for the introduced functionality?

**Comment by lukaszmn** [comment]
Hi @Scrum, what do you mean by test? There is just one example in the project. Do you want me to add another example or enhance the existing one with the new options? Or add actual unit tests?

File: index.js (hunk 1 above)
Comment (Scrum) [reply index.js]: Please correct this and I will merge your solution.

**Comment by Scrum** [comment]
@lukaszmn Hi, oh, sorry, I didn’t notice that there aren’t any tests in them, of course it’s unfair to ask you to add them with your options when expanding.

**Review by Scrum — APPROVED** [review]

**Comment by Scrum** [comment]
@lukaszmn publish v0.9.0

**Pull request merged**